File: back/stock/src/main/java/tn/esprit/stock/entities/QuantiteCommande.java
package tn.esprit.stock.entities;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;


@Entity
@Getter
@Setter
@NoArgsConstructor
public class QuantiteCommande {
    @Id
    @GeneratedValue
    private Long idQuantiteCommande;
    private Integer QuantiteCommande;
    private double montant;

    @ManyToOne
    private Produit produit;
    @ManyToOne
    private BonCommande bonCommande;

}

File: front/G-stock/src/app/models/quantite-commande.ts
import { Produit } from './produit';
import { BonCommande } from './bon-commande';

export interface QuantiteCommande {
  idQuantiteCommande: number;
  quantiteCommande: number;
  montant: number;
  produit: Produit;
  bonCommande: BonCommande | null; // or BonCommande if it can never be null
}
